lib/ui/painting/gradient.cc

```diff
@@ -45,10 +45,14 @@ void CanvasGradient::initLinear(const tonic::Float32List& end_points,
 
   SkPoint p0 = SkPoint::Make(end_points[0], end_points[1]);
   SkPoint p1 = SkPoint::Make(end_points[2], end_points[3]);
-  const DlColor* colors_array = reinterpret_cast<const DlColor*>(colors.data());
+  std::vector<DlColor> dl_colors;
```

Comment:
This is another place where I expect that we'll update the dart:ui API to provide float32 colors.

Reply:
Yep, that will happen in a different PR and have a different integration test for it.  I added a TODO.